src/storage/mod.rs

```diff
@@ -479,10 +480,17 @@ impl<E: Engine, L: LockManager, P: PdClient + 'static> Storage<E, L, P> {
                 {
                     let begin_instant = Instant::now_coarse();
 
+                    let isolation_level = if sample_step > 0 {
+                        limit *= sample_step;
+                        IsolationLevel::Rc
```

Comment:
Why not pass RC directly?

Reply:
Or else we have to modify the MVCC inner scanners and use some parameters to inform them to ignore locks which also introduces some state to them like an inner counter and an ignore-lock-flag.

Reply:
Why not pass RC from TiDB? Please do not implement using HACK.